Failed building wheel for imagecodecs during docker image creation

Hi, 

when creating the docker image on my computer (Ubuntu 16.04.6) I get the following error during step 10/13 (Step 10/13 : RUN pip3 install deeplabcut):
```
> Collecting imagecodecs>=2020.2.18 (from tifffile>=2019.7.26->scikit-image->deeplabcut)
>   Downloading https://files.pythonhosted.org/packages/75/95/3699360d15e5e5d450bbd16ea554fb185117947e76fc772078c4f59ae1c1/imagecodecs-2020.2.18.tar.gz (8.8MB)
> Requirement already satisfied: parso==0.1.1 in /usr/local/lib/python3.6/dist-packages (from jedi>=0.10->ipython->deeplabcut)
> Requirement already satisfied: wcwidth in /usr/local/lib/python3.6/dist-packages (from prompt-toolkit<2.0.0,>=1.0.4->ipython->deeplabcut)
> Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.6/dist-packages (from pexpect; sys_platform != "win32"->ipython->deeplabcut)
> Building wheels for collected packages: pyyaml, easydict, moviepy, proglog, imageio-ffmpeg, imagecodecs
>   Running setup.py bdist_wheel for pyyaml: started
>   Running setup.py bdist_wheel for pyyaml: finished with status 'done'
>   Stored in directory: /root/.cache/pip/wheels/a7/c1/ea/cf5bd31012e735dc1dfea3131a2d5eae7978b251083d6247bd
>   Running setup.py bdist_wheel for easydict: started
>   Running setup.py bdist_wheel for easydict: finished with status 'done'
>   Stored in directory: /root/.cache/pip/wheels/9a/88/ec/085d92753646b0eda1b7df49c7afe51a6ecc496556d3012e2e
>   Running setup.py bdist_wheel for moviepy: started
>   Running setup.py bdist_wheel for moviepy: finished with status 'done'
>   Stored in directory: /root/.cache/pip/wheels/a3/3c/07/45afe2bd5dbd3f935f445545d645f0f8c05d48340136367d45
>   Running setup.py bdist_wheel for proglog: started
>   Running setup.py bdist_wheel for proglog: finished with status 'done'
>   Stored in directory: /root/.cache/pip/wheels/65/56/60/1d0306a8d90b188af393c1812ddb502a8821b70917f82dcc00
>   Running setup.py bdist_wheel for imageio-ffmpeg: started
>   Running setup.py bdist_wheel for imageio-ffmpeg: finished with status 'done'
>   Stored in directory: /root/.cache/pip/wheels/a4/97/d4/c700927e4f2da1db33918b51d6a9eeaabc430f7ed8cb3fb362
>   Running setup.py bdist_wheel for imagecodecs: started
>   Running setup.py bdist_wheel for imagecodecs: finished with status 'error'
>   Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-o11004t_/imagecodecs/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpn5slxnxbpip-wheel- --python-tag cp36:
>   running bdist_wheel
>   running build
>   running build_py
>   creating build
>   creating build/lib.linux-x86_64-3.6
>   creating build/lib.linux-x86_64-3.6/imagecodecs
>   copying imagecodecs/_imagecodecs.py -> build/lib.linux-x86_64-3.6/imagecodecs
>   copying imagecodecs/__init__.py -> build/lib.linux-x86_64-3.6/imagecodecs
>   copying imagecodecs/imagecodecs.py -> build/lib.linux-x86_64-3.6/imagecodecs
>   copying imagecodecs/__main__.py -> build/lib.linux-x86_64-3.6/imagecodecs
>   creating build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-lcms2 -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-bitshuffle -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-charls -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-zlib -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-blosc -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-openjpeg -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-jpeg -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-jpg_0xc3 -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-lz4 -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-libtiff -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-zopfli -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-libpng -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-fastlz -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-brotli -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-libmng -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-libwebp -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-zstd -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-bzip2 -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-zfp -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-libaec -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-snappy -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-liblzma -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-lzf -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-giflib -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-jxrlib -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-libjpeg-turbo -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   copying imagecodecs/licenses/LICENSE-brunsli -> build/lib.linux-x86_64-3.6/imagecodecs/licenses
>   running build_ext
>   building 'imagecodecs._aec' extension
>   creating build/temp.linux-x86_64-3.6
>   creating build/temp.linux-x86_64-3.6/imagecodecs
>   x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Iimagecodecs -I/usr/include/python3.6m -I/tmp/pip-build-o11004t_/imagecodecs/.eggs/numpy-1.18.4-py3.6-linux-x86_64.egg/numpy/core/include -c imagecodecs/_aec.c -o build/temp.linux-x86_64-3.6/imagecodecs/_aec.o
>   In file included from /tmp/pip-build-o11004t_/imagecodecs/.eggs/numpy-1.18.4-py3.6-linux-x86_64.egg/numpy/core/include/numpy/ndarraytypes.h:1832:0,
>                    from /tmp/pip-build-o11004t_/imagecodecs/.eggs/numpy-1.18.4-py3.6-linux-x86_64.egg/numpy/core/include/numpy/ndarrayobject.h:12,
>                    from /tmp/pip-build-o11004t_/imagecodecs/.eggs/numpy-1.18.4-py3.6-linux-x86_64.egg/numpy/core/include/numpy/arrayobject.h:4,
>                    from imagecodecs/_aec.c:598:
>   /tmp/pip-build-o11004t_/imagecodecs/.eggs/numpy-1.18.4-py3.6-linux-x86_64.egg/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
>    #warning "Using deprecated NumPy API, disable it with " \
>     ^
>   imagecodecs/_aec.c:602:20: fatal error: libaec.h: No such file or directory
>   compilation terminated.
>   error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
>   
>   ----------------------------------------
>   Failed building wheel for imagecodecs
>   Running setup.py clean for imagecodecs
> Successfully built pyyaml easydict moviepy proglog imageio-ffmpeg
> Failed to build imagecodecs
> Installing collected packages: msgpack, numpy, msgpack-numpy, tabulate, tensorpack, pyyaml, numexpr, tables, networkx, PyWavelets, imageio, imagecodecs, tifffile, kiwisolver, matplotlib, scikit-image, Shapely, opencv-python, imgaug, easydict, certifi, chardet, intel-openmp, urllib3, idna, requests, proglog, imageio-ffmpeg, moviepy, h5py, patsy, pandas, statsmodels, ruamel.yaml.clib, ruamel.yaml, deeplabcut
>   Found existing installation: numpy 1.14.1
>     Uninstalling numpy-1.14.1:
>       Successfully uninstalled numpy-1.14.1


```


Then it proceeds to install  a bunch of other stuff, and it stops with:



```
> building 'imagecodecs._aec' extension
>     creating build/temp.linux-x86_64-3.6
>     creating build/temp.linux-x86_64-3.6/imagecodecs
>     x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Iimagecodecs -I/usr/include/python3.6m -I/usr/local/lib/python3.6/dist-packages/numpy/core/include -c imagecodecs/_aec.c -o build/temp.linux-x86_64-3.6/imagecodecs/_aec.o
>     In file included from /usr/local/lib/python3.6/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1822:0,
>                      from /usr/local/lib/python3.6/dist-packages/numpy/core/include/numpy/ndarrayobject.h:12,
>                      from /usr/local/lib/python3.6/dist-packages/numpy/core/include/numpy/arrayobject.h:4,
>                      from imagecodecs/_aec.c:598:
>     /usr/local/lib/python3.6/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
>      #warning "Using deprecated NumPy API, disable it with " \
>       ^
>     imagecodecs/_aec.c:602:20: fatal error: libaec.h: No such file or directory
>     compilation terminated.
>     error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
>     
>     ----------------------------------------
> Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-o11004t_/imagecodecs/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-cz_gglay-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-o11004t_/imagecodecs/
> You are using pip version 9.0.1, however version 20.1 is available.
> You should consider upgrading via the 'pip install --upgrade pip' command.
> The command '/bin/sh -c pip3 install deeplabcut' returned a non-zero code: 1

```


Thank you for your support!

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,6 +7,7 @@ RUN pip install imageio-ffmpeg
 FROM bethgelab/deeplearning:cuda9.0-cudnn7
 RUN add-apt-repository ppa:deadsnakes/ppa #ADDS NEW REPO
 RUN add-apt-repository --remove ppa:jonathonf/python-3.6 #REMOVES BROKEN REPO
+RUN apt-get install -y apt-utils
 RUN apt-get update
 RUN apt-get -y install ffmpeg
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -10,8 +10,11 @@ RUN add-apt-repository --remove ppa:jonathonf/python-3.6 #REMOVES BROKEN REPO
 RUN apt-get update
 RUN apt-get -y install ffmpeg
 
-#RUN pip install --upgrade pip #see issue #25
+
 RUN pip install tensorflow-gpu==1.8
+#see issue #25:
+RUN pip install --upgrade pip 
+
 RUN pip install ruamel_yaml
 RUN pip3 install deeplabcutcore
 RUN pip install ipywidgets

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,12 +7,12 @@ RUN pip install imageio-ffmpeg
 FROM bethgelab/deeplearning:cuda9.0-cudnn7
 RUN add-apt-repository ppa:deadsnakes/ppa #ADDS NEW REPO
 RUN add-apt-repository --remove ppa:jonathonf/python-3.6 #REMOVES BROKEN REPO
-RUN apt-get install -y apt-utils
 RUN apt-get update
 RUN apt-get -y install ffmpeg
 
 #RUN pip install --upgrade pip #see issue #25
 RUN pip install tensorflow-gpu==1.8
+RUN pip install ruamel_yaml
 RUN pip3 install deeplabcutcore
 RUN pip install ipywidgets
 RUN pip3 install ipywidgets

